Added branch guidance (#5614)
microsoft/fast: CONTRIBUTING.md

### Submitting a pull request

> If you'd like to contribute by fixing a bug, implementing a feature, or even correcting typos in our documentation, you'll want to submit a pull request. Before submitting a pull request, be sure to [rebase](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) your branch (typically from master) or use the *merge* button provided by GitHub.

Author: If you use the merge button provided from GitHub, it should squash merge per the settings for this repository, and rebasing is not necessary. If there are conflicts that prevent the commit from being added by rebasing, the repository will not let you merge your branch.

If you are addressing multiple issues which are unrelated, consider either doing multiple pull requests, or generate separate change files to ensure accurate generation of changelogs and versioning of packages.
:::

:::note

> If you are finding that your changes are either breaking changes or require multiple pull requests, open an issue to discuss this.

Reviewer: Suggestion to replace "open an issue to discuss this" with "open a [discussion](https://github.com/microsoft/fast/discussions)".

> If you are finding that your changes are either breaking changes or require multiple pull requests, open an issue to discuss this.

Author: ```suggestion
If you are finding that your changes are either breaking changes or require multiple pull requests, open a [discussion](https://github.com/microsoft/fast/discussions) to discuss this.
```
